--- a/bin/mpicxx	2021-02-04 18:15:11.233632360 +0100
+++ b/bin/mpicxx	2021-02-05 09:36:21.396922569 +0100
@@ -50,7 +50,7 @@
 if [ x"$opt_args" == x"" ]; then
     case "${compiler_short_name}" in
     icc|icpc|dpcpp)   "$dir"/mpiicpc -cxx=$compiler_name "$@" ;;
-    *g++*)        "$dir"/mpigxx -cxx=$compiler_name "$@" ;;
+    *g++*|clang*++|mcxx|echo) "$dir"/mpigxx -cxx=$compiler_name "$@" ;;
     mpicxx)     "$dir"/mpigxx "$@" ;;
     *)  
             echo "Error: unsupported compiler name '$compiler_name'."
@@ -60,7 +60,7 @@
 else
     case "${compiler_short_name}" in
     icc|icpc|dpcpp)   "$dir"/mpiicpc -cxx=$compiler_name "$@" $opt_args ;;
-    *g++*)        "$dir"/mpigxx -cxx=$compiler_name "$@" $opt_args ;;
+    *g++*|clang*++|mcxx|echo) "$dir"/mpigxx -cxx=$compiler_name "$@" $opt_args ;;
     mpicxx)     "$dir"/mpigxx "$@" $opt_args ;;
     *)  
             echo "Error: unsupported compiler name '$compiler_name'."
