--- a/opt/intel/compilers_and_libraries_2020.1.217/linux/mpi/intel64/bin/mpicc	2020-06-30 14:22:04.510566478 +0200
+++ b/opt/intel/compilers_and_libraries_2020.1.217/linux/mpi/intel64/bin/mpicc	2020-06-30 14:24:58.250998988 +0200
@@ -50,7 +50,7 @@
 if [ x"$opt_args" == x"" ]; then
     case "${compiler_short_name}" in
     icc)    $dir/mpiicc -cc=$compiler_name "$@" ;;
-    cc|*gcc*) $dir/mpigcc -cc=$compiler_name "$@" ;;
+    cc|*gcc*|mcc|clang) $dir/mpigcc -cc=$compiler_name "$@" ;;
     mpicc)  $dir/mpigcc "$@" ;;    
     *)  
             echo "Error: unsupported compiler name '$compiler_name'."
@@ -60,7 +60,7 @@
 else
     case "${compiler_short_name}" in
     icc)    $dir/mpiicc -cc=$compiler_name "$@" $opt_args ;;
-    cc|*gcc*) $dir/mpigcc -cc=$compiler_name "$@" $opt_args ;;
+    cc|*gcc*|mcc|clang) $dir/mpigcc -cc=$compiler_name "$@" $opt_args ;;
     mpicc)  $dir/mpigcc "$@" $opt_args ;;    
     *)  
             echo "Error: unsupported compiler name '$compiler_name'."
